packages/variable-resolver/src/browser/variable-resolver-service.ts

```diff
@@ -50,6 +50,9 @@ export class VariableResolverService {
     }
 
     protected doResolve(value: Object | undefined, context: VariableResolverService.Context): Object | undefined {
+        if (value === undefined || value === null) {
```

Comment:
Although `null` is an `object`, we might want to return with `undefined` if the `value` is `null` instead of `null`.

Something like this:
```js
if (value === undefined || value === null) {
  return undefined;
}
```

What do you think? We try to avoid `null` if possible, at least that was the [policy](https://github.com/theia-ide/theia/wiki/Coding-Guidelines#null-and-undefined) we tried to follow.

Reply:
Hmm, is changing the type a concern of a variable resolver? If deemed necessary, I'm happy to make the change.

Reply:
`null` check should be removed, we don't allow `null` usages in APIs and don't expect anybody to pass it

Reply:
Problem is, this guard was added in order to successfully parse an external object tree which contains nulls:

https://github.com/Marus/cortex-debug/blob/master/package.json#L196

As null is valid in json, I think this check should remain. Is that OK?